libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -604,6 +608,56 @@ void AP_DDS_Client::update_topic(geographic_msgs_msg_GeoPointStamped& msg)
 }
 #endif // AP_DDS_GPS_GLOBAL_ORIGIN_PUB_ENABLED
 
+#if AP_DDS_STATUS_PUB_ENABLED
+bool AP_DDS_Client::update_topic(ardupilot_msgs_msg_Status& msg)
+{
+    // Fill the new message.
+    const auto &vehicle = AP::vehicle();
+    const auto &battery = AP::battery();
+    msg.vehicle_type = static_cast<uint8_t>(AP::fwversion().vehicle_type);
+    msg.armed = hal.util->get_soft_armed();
+    msg.mode = vehicle->get_mode();
+    msg.flying = vehicle->get_likely_flying();
+    msg.external_control = true; // Always true for now. To be filled after PR#28429.
+    uint8_t fs_iter = 0;
+    msg.failsafe_size = 0;
+    if (AP_Notify::flags.failsafe_radio) {
```

Comment:
Can we add another comment that "TODO move from AP_Notify to vehicle functions"

Reply:
`virtual bool in_rc_failsafe() const { return true; }`

Reply:
Check the radio failsafe using this call rather than AP_Notify:
https://github.com/ArduPilot/ardupilot/blob/9f29606d1c915f0dd64a06babf628c623311783a/libraries/RC_Channel/RC_Channel.h#L517

The devs do not want us using notify unless there is no other option.